code/modules/overmap/ships/engines/ion_thruster.dm

```diff
@@ -0,0 +1,86 @@
+/datum/ship_engine/ion
+	name = "ion thruster"
+	var/obj/machinery/ion_engine/thruster
+
+/datum/ship_engine/ion/New(var/obj/machinery/_holder)
+	..()
+	thruster = _holder
+
+/datum/ship_engine/ion/Destroy()
+	thruster = null
+	. = ..()
+
+/datum/ship_engine/ion/get_status()
+	return thruster.get_status()
+
+/datum/ship_engine/ion/get_thrust()
+	return thruster.get_thrust()
+
+/datum/ship_engine/ion/burn()
+	return thruster.burn()
+
+/datum/ship_engine/ion/set_thrust_limit(var/new_limit)
+	thruster.thrust_limit = new_limit
+
+/datum/ship_engine/ion/get_thrust_limit()
+	return thruster.thrust_limit
+
+/datum/ship_engine/ion/is_on()
+	return thruster.on && thruster.powered()
+
+/datum/ship_engine/ion/toggle()
+	thruster.on = !thruster.on
+
+/datum/ship_engine/ion/can_burn()
+	return thruster.on && thruster.powered()
+
+/obj/machinery/ion_engine
+	name = "ion propulsion device"
+	desc = "An advanced ion propulsion device, using energy and minutes amount of gas to generate thrust."
+	icon = 'icons/turf/shuttle_parts.dmi'
+	icon_state = "nozzle"
+	power_channel = ENVIRON
+	idle_power_usage = 100
+	anchored = TRUE
+	// construct_state = /decl/machine_construction/default/panel_closed
+	var/datum/ship_engine/ion/controller
+	var/thrust_limit = 1
+	var/on = 1
+	var/burn_cost = 7500
```

Comment:
Ion thrusters don't actually use gas in this (Bay's) implementation. It's abstracted away. It's basically just an upgrade to gas engines in terms of fuel usage (in that it uses none), but uses significant amounts of power. I know you're a rocket nerd but I'd suggest not adding them using gas, because half the point is that it's an upgrade where you don't need to pipe gas to them so you're sorta messing with it from a gameplay balance perspective.

Strangely, Ion Engines were just better in every way on Bay, they used LESS power than rocket engines while producing reasonable thrust and not using fuel. Rocket engines also used such gargantuan amounts of power that an APC could scarcely hope to keep up with them.

I rebalanced rocket engines to use less power, but use fuel, and ion engines to use more power, but no fuel.